### DISCUSSION TITLE
Disabling fields

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#682] [Q&A] [answered]
Hi!

I know I can build a compiler for that but, since Unpoly already has `up-switch`/`up-show-for` for something similar, would it be a possibility to have a `up-switch` equivalent for toggling fields enable/disabled instead of hiding?

TIA

**Comment by hpvd** [comment]
+1 on this.

**Comment by triskweline** [comment]
I agree this would be useful.

Can we think offer other switching effects, aside from hiding and disabling?

**Reply by hpvd** [reply]
> Can we think offer other switching effects, aside from hiding and disabling?

just to name some use-cases:
hiding and disabling are the ones we are extensively using for depend fields/options in lots of larger webforms...
and disabling is also used in some menus (as teasers/user action trigger).

Its always pretty handy if its possible
 - individually per field/option/item
 - but also globally for a form e.g. during debugging by switching all depend fields from production setting `hidden` (invisible) to `disabled` (and with this visible)...

**Reply by alexandremjacques** [reply]
>but also globally for a form e.g. during debugging by switching all depend fields from production setting hidden (visible) to disabled (and with this visible)...

I guess this is covered by the `up-disable` on the `<form>`, right?

>Can we think offer other switching effects, aside from hiding and disabling?

I don't really use effects (I find them distracting). If it's something that people use, I guess it would be ok.

My uses cases are quite simple: I need to toggle the `disabled` attribute for fields based on the value for another field. I achieved that with (I know I'm not supposed to use the `up-*` prefix 😁):

```
up.on('change', '[up-toggle]', function(element) {
    let elementValue = element.target.value
    let targetSelector = element.target.getAttribute('target')
    let targets = document.querySelectorAll(targetSelector)

    targets.forEach(target => {
        target.disabled = target.getAttribute('up-enable-for') !== elementValue;
    })
})
```

My guess is that there is a better way to do that but its working fine for what I need right now.

**Reply by triskweline** [reply]
> Can we think offer other switching effects, aside from hiding and disabling?

I didn't mean visual effects. I meant if there are other things one would want to couple to an `[up-switch]` toggle, aside from toggling visibility or being disabled. Like expanding accordions, toggling classes, etc.

**Reply by hpvd** [reply]
some more use-case details... maybe this way its easier to understand/decide what makes sense in this field:

- we have groups of fields placed on different tabs
  - on the bottom of a tab there is a button to jump to (open) next tab
  - we use lazy validations for some fields (its done at the end before finalizing /sending the form): as the result, in false case, we open the tab with the first false validation and auto-scroll to the position on the tab
- on narrow screens we switch to accordions instead of tabs which uses the same logic as the tabs
- for debugging expanding all accordions helps again...

**Reply by hpvd** [reply]
or maybe things from the a11y field where user preferences are stored in a user profil and read and applied on login or lazy from cookie?

like switching between
- two different "animation themes" (while storing the setting in user profile)
  - standard and
  - with "reduced motion"
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
- light/dark mode
- standard and high contrast theme
- ...

**Comment by triskweline** [comment, marked as answer]
[Version 3.1.1](https://github.com/unpoly/unpoly/discussions/762) now has `[up-enable-for]` and `[up-disable-for]`. You can also implement your own custom switching effects, by listening to `up:form:switch`.

See [Switching form date](https://unpoly.com/switching-form-state) for a longer guide.
